subprojects/wrapper/src/integTest/groovy/org/gradle/integtests/WrapperLoggingIntegrationTest.groovy

```diff
@@ -31,4 +34,39 @@ task emptyTask
         then:
         result.output.empty
     }
+
+    def "wrapper logs when there is a problem setting permissions"() {
```

Comment:
:o: It might be easier to test "setting the permission" as a unit test. Then you could just Mock the interface that calls off to `ProcessBuilder` and respond with any kind of error condition.

Reply:
Don't feel very strongly about it. I think it would just be easier to test but it would require additional refactoring of the production source code. It's OK for now.